### PR TITLE
EBS Test Execution Plan with Vagrant

### PR DESCRIPTION
**Pull request by akutz**
This patch adds a Vagrant file for executing EBS end-to-end tests with REX-Ray and libStorage.

### CODE REVIEW

File: drivers/storage/ebs/tests/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+$aws_secgrp = "pbs"
+$aws_ami    = "ami-de347abe" # Amazon Linux x64 PV EBS
+$aws_cnxn =   Fog::Compute.new(
+    :provider => :aws,
+    :region => $aws_region,
+    :aws_access_key_id => $aws_akey,
+    :aws_secret_access_key => $aws_skey)
+
+# ssh config
+$ssh_pkey   = "/Users/akutz/.ssh/ec2-akutz.pem"
```
Comment (codenrhoden) [inline drivers/storage/ebs/tests/Vagrantfile]: Do you want your PEM hardcoded in here? What if it was generic, e.g. `~/.ssh/ec2.pem` but overridable via ENV?

File: drivers/storage/ebs/tests/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+require 'fog'
+
+# settings
+$autostart_clients = true
+
+# aws config
+$aws_akey   = ENV['AWS_AKEY']
+$aws_skey   = ENV['AWS_SKEY']
+$aws_region = "us-west-1"
+$aws_kpname = "akutz"
```
Comment (codenrhoden) [inline drivers/storage/ebs/tests/Vagrantfile]: Might want this to be an ENV var as well.

File: drivers/storage/ebs/tests/Vagrantfile (hunk 1 above)
Comment (akutz) [inline drivers/storage/ebs/tests/Vagrantfile]: Agreed. This was a PoC to get it working.

File: drivers/storage/ebs/tests/Vagrantfile (hunk 2 above)
Comment (akutz) [inline drivers/storage/ebs/tests/Vagrantfile]: Good call.

**Comment by akutz** [comment]
FYI @codenrhoden, I may merge this with some of the above suggestions outstanding once I get the second client node working. I just wanted to PoC the thing. I intend to add a Readme to make it clear which Vagrant plug-ins I installed as well.

File: drivers/storage/ebs/tests/README.md [hunk 3]
```diff
@@ -1,9 +1,134 @@
...
+vagrant plugin install vagrant-hostmanager
 ```
-[default]
-aws_access_key_id=AKIAIOSFODNN7EXAMPLE
-aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+
+### Test Plan Settings
+The following environment variables may be used to configure the `Vagrantfile`.
+Please note that use of the default AWS environment variables was avoided. This
+is to refrain from affecting the unit/integration tests as they two may make
```
Comment (codenrhoden) [inline drivers/storage/ebs/tests/README.md]: "as they two may make"...  I think this intended to be "as they may make..." ?

File: drivers/storage/ebs/tests/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,487 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+require 'fileutils'
+require 'shellwords'
+require 'fog'
```
Comment (codenrhoden) [inline drivers/storage/ebs/tests/Vagrantfile]: Is fog included by default? Or is it a test dependency that needs to be installed?

File: drivers/storage/ebs/tests/README.md (hunk 3 above)
Comment (akutz) [inline drivers/storage/ebs/tests/README.md]: Thanks for catching this.

**Pull request merged**